test/integration/storages_state_test.lua

```diff
@@ -0,0 +1,89 @@
+local fio = require('fio')
+
+local t = require('luatest')
+
+local helpers = require('test.helper')
+
+local fiber = require("fiber")
+
+local pgroup = t.group('replicas_state', {
+    {engine = 'memtx'}
+})
+
+local all_storages_initialized = false
+
+local function wait_storages_init(g)
+    local storages_initialized = false
+    local attempts_left = 5
+    local wait_for_init_timeout = 1
+    while (attempts_left > 0 and not storages_initialized) do
+        local results, err = g.cluster.main_server.net_box:call("crud.get_storages_state", {})
+        t.assert_equals(err, nil, "Error getting storags states")
+        storages_initialized = true
+        for _,v in pairs(results) do
+            if not v.storage_initialized then
+                storages_initialized = false
+            end
+        end
+        if not storages_initialized then
+            fiber.sleep(wait_for_init_timeout)
+            attempts_left = attempts_left-1
+        end
+    end
+    return storages_initialized
+end
+
+pgroup.before_all(function(g)
+    g.cluster = helpers.Cluster:new({
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint('srv_select'),
+        use_vshard = true,
+        replicasets = helpers.get_test_replicasets(),
+        env = {
+            ['ENGINE'] = g.params.engine,
+        },
+    })
+    g.cluster:start()
+
+    -- wait for storages to initialize
+    all_storages_initialized = wait_storages_init(g)
```

Comment:
Do we need it here? Isn't luatest wait for instance roles init?

Reply:
Seems like it is
https://github.com/tarantool/cartridge/blob/9ea29aed1a480df0ebb4bf42b0bc0c5903a24023/cartridge/test-helpers/cluster.lua#L338-L346

Reply:
Yes you are right, cluster start waits for roles configured. I wanted to rely on this also, but... I need this code because sometimes I get "peer closed" error for some connections right after the cluster start. Retry is required to give a small timeout to reconnect. As I can see in logs, disconnect and reconnect always occur right after cluster start in all tests. And in my case reconnect does not happen fast enough sometimes.
Suppose, we should investigate this behavior. But I can do nothing with this from CRUD side.

Reply:
Let's just add your current observations into a comment in the code and move beyond. It is fair to let future readers know why the function is here.